rfcs/rfc-fas-search-in-sbc-auth.md

```diff
@@ -0,0 +1,118 @@
+# RFC For Including FAS [Fee Accounting System] dashboard in Auth application's staff dashboard
+
+- Start Date: (2021-09-15)
+- Target Major Version: None
+- Reference Issues: None
+- Entity Issue: https://github.com/bcgov/entity/issues/8510
+- Implementation PR: None
+
+# Summary
+
+Embed Fas Search component in SBC-Auth project so that  staff can avail FAS Routing slip search functionality without navigating to FAS system.
+ 
+# Basic example
+
+Building as search component can see below 
+https://github.com/shabeeb-aot/fas-ui/blob/feature/8605/src/lib-setup.js
+
+
+# Motivation
+
+This FAS search component is developed with in the fas application. The requirement is to host the search page with in the auth application as well with minimal changes in the auth-web[hosting application].
+With this approach , the major benefit is that the component still lives with in FAS application which makes maintainence easier for developers rather than maintaining in a different shared library repo.This approach can be adapted incase if another dashboards ie PPR dashboard has to be hosted in auth application. The component still lives with in the original application as well as it can be used in another project. More dadvantges are listed below
+ 
+1) Reuse existing FAS Routing slips search functionality in SBC Auth or any project needed
+
+2) Eliminate duplicate code / fix bugs in one place only
+
+3) code once, use it in multiple place
+
+4) Easier component maintainability
+
+6) Pathway for other similar use cases where one applications dashboard have to be used in another application.
+
+
+# Detailed design
+
+**How to build the components**
+
+The components will be developed in same fas repo https://github.com/bcgov/fas-ui/. Only search component will build as a vue plugin.
+
+https://cli.vuejs.org/guide/build-targets.html#library 
+
+In short:
+1. use same Fas github repo
+2. Check it out locally
+3. build as library using npm run build:lib
+4. it will generate lib folder.
+5. Commit/publish to GitHub (*)
+
+No changes needed to FAS project
+
+
+
+**Hosting**
+
+The components will be on github only and use npm install https://github.com/bcgov/fas-ui/. This will install in code base with umd minified file and search.scss only.
+
+**How to use it**
+
+Just use it like any other npm package. 
+install using below comment from git
+```
+npm i https://github.com/bcgov/fas-ui.git --save
+```
+or include in package.json
+```
+ "fas-ui": "git+https://github.com/bcgov/fas-ui.git",
+```
+![sbc- auth package.json image](rfc-fas-search-in-sbc-auth/package.png)
```

Comment:
How sensitive is this component (app) to various imports (Vue, axios, etc)? Does anything have to be synchronized, or are the dependencies automatically handled (or don't matter)?

Reply:
Its the same inherent behaviour like any other plugins, It will have its own dependence if it's not available in parent application it will automatically installed as dependency. NPM's module system will manage it.  When there is breaking change in any imports like vue 2 and vue 3 then we have to handle it (this will be the same case for all plugins). Other wise,  all the  dependencies are automatically handled.